packages/revocation/src/implementations/statuslistentry-verification.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  CredentialSubject,
+  StatusList2021Entry,
+  VerifiableCredential,
+} from '@ew-did-registry/credentials-interface';
+import * as pako from 'pako';
+import axios from 'axios';
+
+/**
+ * A Class to verify the StatusList2021Entry
+ */
+export class StatusListEntryVerification {
+  private verify: (vc: string, proof_options: string) => Promise<any>;
+
+  constructor(
+    callbackFunction: (vc: string, proof_options: string) => Promise<any>
+  ) {
+    this.verify = callbackFunction;
+  }
+
+  /**
+   * Verifies the status of StatusListEntry
+   * @param statusListEntry which needs to be verified
+   */
+  async verifyStatusList(statusListEntry: StatusList2021Entry) {
+    if (statusListEntry.type !== 'StatusList2021Entry') {
+      throw new Error('Not a StatusList2021Entry type');
+    }
+    if (statusListEntry.statusPurpose !== 'revocation') {
+      throw new Error('StatusList2021Entry does not have revocation purpose');
+    }
+    const credential = await this.fetchCredential(
+      statusListEntry.statusListCredential
+    );
+    // TODO add checks for the returned StatusListCredential, type not confirmed for now
+    // if (!credential.@context.contains(StausList2021ContextURI)) {
+    //   throw new Error(
+    //     'Missing expected context URI for credential using StatusList2021'
+    //   );
+    // }
+    // if (!credential.issuer) {
+    //   throw new Error('No issuer found for the credential');
+    // }
+    const verifyResults = await this.verify(
+      JSON.stringify(credential),
+      JSON.stringify({})
+    );
+
+    if (verifyResults.errors.length) {
+      throw new Error(
+        `StatusListCredential is invalid, error: ${verifyResults.errors.join(
+          ','
+        )}`
+      );
+    }
+    return true;
+  }
+
+  /**
+   * Validates the URL
+   * @param url to be validated
+   * @returns true if URL is valid
+   */
+  private isValidStatusListCredentialURI(url: string) {
+    const splittedUrl = url.split(':');
+    console.log(splittedUrl[0]);
+    if (splittedUrl[0] == 'https') {
+      return true;
+    }
+    throw new Error('Not a https url');
+  }
+
+  /**
+   * Fetches the StatusListCredential
+   * @param statusListCredential URL to fetch from
+   * @returns
+   */
+  async fetchCredential(statusListCredentialUrl: string) {
```

Comment:
Can this be private?
```suggestion
  async private fetchCredential(statusListCredentialUrl: string) {
```